### PR TITLE
Feature gate macro_reexport. Fixes #20906

### LINKED ISSUE #20906: Feature-gate #[macro_reexport]
This feature was only designed with libstd's use case in mind, and could use additional thought. My understanding is that the compiler/stdlib bootstrap circumvents stability enforcement in such a way that we could use this in libstd even with the feature gate.

Should do this before 1.0-beta1 if at all.

### PR DESCRIPTION
**Pull request by flaper87**

### CODE REVIEW

File: src/test/auxiliary/macro_reexport_2.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 #![crate_type = "dylib"]
+#[feature(macro_reexport)]
```
Comment (huonw) [inline src/test/auxiliary/macro_reexport_2.rs]: Should these be `#!`?

Comment (FlaPer87) [inline src/test/auxiliary/macro_reexport_2.rs]: yes, it should. forgot to update these cases.

**Comment by Manishearth** [comment]
This will break currently, feature gate tuples now have an additional "1.0.0" argument. (Fixed in my rollup)

**Pull request merged**